cmd/server/helper_keys.go

```diff
@@ -18,7 +18,7 @@ func createRS256KeysIfNotExist(c *config.Config, set, kid, use string) {
 	if _, err := ctx.KeyManager.GetKey(set, kid); errors.Cause(err) == pkg.ErrNotFound {
 		logrus.Infof("Key pair for signing %s is missing. Creating new one.", set)
 
-		keys, err := generator.Generate("")
+		keys, err := generator.Generate(set)
```

Comment:
I think I wasn't able to get the point across yet :D 

When booting Hydra on a new database, the signing keys for OIDC will be generated with:

```
{
set: "openid",
keys: [{kid: "private"}, {kid: "public"}]
}
```

In this case, the kids private and public are unique because the set does not have any other keys. Let's say the administrator of Hydra now adds a new keys to the set by accident:


```
{
set: "openid",
keys: [{kid: "private"}, {kid: "public"}, {kid: "public:just-any-key-whatever"}]
}
```

The private and public keys are still unique in this case - so no problem. Let's say the admin now wants to do some fancy key rotation. First, he renames the keys private and public:

```
{
set: "openid",
keys: [{kid: "private:old"}, {kid: "public:old"}]
}
```

next he adds the new keys:

```
{
set: "openid",
keys: [
  {kid: "private"}, {kid: "public"},
  {kid: "private:old"}, {kid: "public:old"}
]
}
```

Now, we have four keys. The old ones plus the fresh ones from key rotation. In a later stage I want to provide automatic key rotation, so this process would be automated by, for example, calling `PUT /keys/rotate/openid` with the result being something like this:


```
{
set: "openid",
keys: [
  {kid: "private"}, {kid: "public"},
  {kid: "private:some-uuid-or-a-timestamp-or-something-else"}, {kid: "public:some-uuid-or-a-timestamp-or-something-else"}
]
}
```

Does this make more sense now?